### PR TITLE
Add search in the whole json

### PR DESCRIPTION
**Pull request by Saphyel**

### CODE REVIEW

File: src/Context/RestContext.php [hunk 1]
```diff
@@ -241,4 +241,17 @@ public function theRestHeaderShouldExistBe($header, $contents)
...
     }
+
+   /**
+   * @Then I should see :text in the JSON
+   */
+  public function iShouldSee($text)
+  {
+      $responseData = $this->_response->getBody();
+      $responseData = \GuzzleHttp\json_decode($responseData, true);
+      $result = json_encode($responseData);
```
Comment (NoelLH) [inline src/Context/RestContext.php]: What's the rationale for decoding then immediately encoding here?

File: src/Context/RestContext.php [hunk 2]
```diff
@@ -241,4 +241,17 @@ public function theRestHeaderShouldExistBe($header, $contents)
                 ' (actual: '.$header1[0].')');
         }
     }
+
+   /**
+   * @Then I should see :text in the JSON
+   */
+  public function iShouldSee($text)
```
Comment (NoelLH) [inline src/Context/RestContext.php]: I think the method name should match the scenario description, i.e. both should say they're JSON-specific or neither should (and this should accurately match the implementation)

**Comment by Saphyel** [comment]
@NoelLH responseShouldContainExpected is to check that in X path exist Y value, but *we* want to check that a value exist without care about where belongs that value.

**Comment by Saphyel** [comment]
Example of the payload:
```
[{"link":{"weight":"-2","title":"Home","description":null,"menu_name":"main","provider":"menu_link_content","parent":"","enabled":true,"expanded":false,"resettable":false,"translatable":false,"deletable":true,"route_name":"\u003Cfront\u003E","route_parameters":[],"url":"\/","options":[],"meta_data":{"entity_id":"10"},"delete_route":{},"edit_route":{}},"has_children":false,"depth":1,"in_active_trail":false,"subtree":[],"count":1},{"link":{"weight":"0","title":"Fundraise (Landing)","description":null,"menu_name":"main","provider":"menu_link_content","parent":"","enabled":true,"expanded":false,"resettable":false,"translatable":false,"deletable":true,"route_name":"entity.node.canonical","route_parameters":{"node":"22"},"url":"\/fundraise","options":[],"meta_data":{"entity_id":"11"},"delete_route":{},"edit_route":{}},"has_children":false,"depth":1,"in_active_trail":false,"subtree":[],"count":1},{"link":{"weight":"1","title":"What\u0027s going on","description":null,"menu_name":"main","provider":"menu_link_content","parent":"","enabled":true,"expanded":false,"resettable":false,"translatable":false,"deletable":true,"route_name":"entity.node.canonical","route_parameters":{"node":"17"},"url":"\/news","options":[],"meta_data":{"entity_id":"12"},"delete_route":{},"edit_route":{}},"has_children":false,"depth":1,"in_active_trail":false,"subtree":[],"count":1},{"link":{"weight":"2","title":"FAQ","description":null,"menu_name":"main","provider":"menu_link_content","parent":"","enabled":true,"expanded":false,"resettable":false,"translatable":false,"deletable":true,"route_name":"entity.node.canonical","route_parameters":{"node":"1"},"url":"\/frequently-asked-questions","options":[],"meta_data":{"entity_id":"13"},"delete_route":{},"edit_route":{}},"has_children":false,"depth":1,"in_active_trail":false,"subtree":[],"count":1},{"link":{"weight":"3","title":"Legal","description":null,"menu_name":"main","provider":"menu_link_content","parent":"","enabled":true,"expanded":false,"resettable":false,"translatable":false,"deletable":true,"route_name":"entity.node.canonical","route_parameters":{"node":"2"},"url":"\/legal","options":[],"meta_data":{"entity_id":"14"},"delete_route":{},"edit_route":{}},"has_children":false,"depth":1,"in_active_trail":false,"subtree":[],"count":1}]
```

**Pull request merged**